files/project_export.sh

```diff
@@ -118,10 +118,14 @@ dcs(){
           )' > ${PROJECT}/dc_${dc}.json
     if [ !$(cat ${PROJECT}/dc_${dc}.json | jq '.spec.triggers[].type' | grep -q "ImageChange") ]; then
       for container in $(cat ${PROJECT}/dc_${dc}.json | jq -r '.spec.triggers[] | select(.type == "ImageChange") .imageChangeParams.containerNames[]'); do
-        echo "Patching DC..."
         OLD_IMAGE=$(cat ${PROJECT}/dc_${dc}.json | jq --arg cname ${container} -r '.spec.template.spec.containers[] | select(.name == $cname)| .image')
         NEW_IMAGE=$(cat ${PROJECT}/dc_${dc}.json | jq -r '.spec.triggers[] | select(.type == "ImageChange") .imageChangeParams.from.name // empty')
-        sed -e "s#$OLD_IMAGE#$NEW_IMAGE#g" ${PROJECT}/dc_${dc}.json >> ${PROJECT}/dc_${dc}_patched.json
+        if [ -n "$OLD_IMAGE" -a -n "$NEW_IMAGE" ]; then
```

Comment:
I believe I finally understand what's going on here.

If a Pod template in a DeploymentConfig has multiple containers in it, but not all container images are covered by an ImageChange trigger then we should not patch the DeploymentConfig, because `NEW_IMAGE` is empty.

If this is correct please add a comment somewhere, so that future generations can easier understand what's going in here.

Reply:
The issue this PR fixes is actually the opposite: The backup fails if there is a ImageChange trigger that references an inexistent container, thus `OLD_IMAGE` being empty. I added the check for `NEW_IMAGE` as well so we never patch container images when the new image is absent (e.g. a broken trigger spec).

The problem you're describing should never occur, as this script iterates over the ImageChange triggers, not over the containers.

Reply:
@tongpu I added some comments explaining the issue.

Reply:
> The problem you're describing should never occur, as this script iterates over the ImageChange triggers, not over the containers.

I see. I haven't read the code well enough and simply assumed that a loop variable called `container` would loop over containers definitions in a pod.